src/foam/mlang/mlang.js

```diff
@@ -4715,24 +4716,39 @@ foam.CLASS({
       name: 'partialEval',
       type: 'foam.mlang.Expr',
       javaCode: `
-        List<Double> list = new ArrayList<>();
+        if ( getArgs().length == 0 ) return this;
+        if ( getArgs().length == 1 ) return getArgs()[0].partialEval();
+
+        List<Double> valList = new ArrayList<>();
+        List<Expr>   argList = new ArrayList<>();
         for ( var arg : getArgs() ) {
           arg = arg.partialEval();
           if ( arg instanceof Constant ) {
             var value = ((Number) arg.f(this)).doubleValue();
-            list.add(value);
+            valList.add(value);
+          } else {
+            argList.add(arg);
           }
         }
 
-        if ( list.size() == getArgs().length ) {
-          var result = list.stream().reduce(this::reduce).get();
+        var result = valList.stream().reduce(this::reduce);
+        if ( result.isPresent() ) {
+          var value = result.get();
+          if ( ! Double.isFinite(value) ) return new Constant(value);
+          else if ( argList.isEmpty()   ) return new Constant(getRounding() ? Math.round(value) : value);
```

Comment:
No need for 'else' if you've already returned in the 'if'.

Reply:
Can you explain the logic here?

Reply:
There is a case when the result of reduce is a finite number and argList is not empty so it would combine the result with the argList.

For example, `Add(1,2,3,4,5, Transaction.AMOUNT, 6,7,8,9)` will become `Add(Transaction.AMOUNT, 45)` since the result of the reduce is 45 and argList=[Transaction.AMOUNT]

Reply:
Updated